src/main/java/org/simdjson/SimdJsonParser.java

```diff
@@ -34,7 +34,8 @@ public JsonValue parse(byte[] buffer, int len) {
     }
 
     private byte[] padIfNeeded(byte[] buffer, int len) {
-        if (buffer.length - len < PADDING) {
+        if (buffer.length - len < PADDING && len < capacity) {
+            byte[] paddedBuffer = new byte[len + PADDING];
```

Comment:
The reason for maintaining the `paddedBuffer` all the time, regardless of whether it is necessary or not, is to avoid allocations on hot paths. However, I see at least two issues with padding in general. Firstly, it requires adding this extra branch. Secondly, it complicates the API: on one hand, the user doesn't need to be aware of it, but on the other hand, if they want to achieve the best performance, they should pad the input. Therefore, I've been considering removing the need for padding altogether. It should be possible, although I haven't thoroughly researched this topic.

To summarize: I'd start by verifying if removing the padding is possible. If so, I'd remove it and test the performance of the parser. If there is no regression compared to the current version with padding, we have a win-win situation.

Reply:
I think by design, the padding is 64 bytes. However, the 'paddedBuffer' is 34MB, that's such a waste. I'm just change the padding size to 64 bytes.

Reply:
I agree with you that this is a waste. However, in your approach, you are potentially allocating a new array on every call of the `parse` method, which can be costly. 

I've been working on removing the padding entirely. It's a bit complicated, but we will see if it is feasible. I'll report back.